src/udiskslinuxprovider.c

```diff
@@ -355,13 +366,10 @@ on_etc_udisks2_dir_monitor_changed (GFileMonitor     *monitor,
       event_type == G_FILE_MONITOR_EVENT_CHANGES_DONE_HINT)
     {
       gchar *filename = g_file_get_basename (file);
+      gchar *id = dup_id_from_config_name (filename);
       if (g_str_has_suffix (filename, ".conf"))
```

Comment:
Oh I now see what you mean. Yeah, this needs to be fixed. I think it's good to have a separate function for this, but it should first check the given string's length plus it should only be called inside the ``if`` block.